src/app/modules/main/profile_section/profile/models/profile_save_data.nim

```diff
@@ -0,0 +1,51 @@
+import json, strutils, sequtils
+
+include app_service/common/json_utils
+include app_service/common/utils
+
+import app_service/service/profile/dto/profile_showcase_preferences
+
+type ProfileShowcaseSaveEntry* = ref object of RootObj
+  key*: string
+  showcaseVisibility*: ProfileShowcaseVisibility
+
+type ProfileShowcaseSaveSocialLink* = ref object of RootObj
+  url*: string
+  text*: string
+  showcaseVisibility*: ProfileShowcaseVisibility
+
+type ProfileShowcaseSaveData* = ref object of RootObj
+  communities*: seq[ProfileShowcaseSaveEntry]
+  accounts*: seq[ProfileShowcaseSaveEntry]
+  collectibles*: seq[ProfileShowcaseSaveEntry]
+  assets*: seq[ProfileShowcaseSaveEntry]
+  socialLinks*: seq[ProfileShowcaseSaveSocialLink]
+
+proc toProfileShowcaseSaveEntry*(jsonObj: JsonNode): ProfileShowcaseSaveEntry =
+  result = ProfileShowcaseSaveEntry()
+  discard jsonObj.getProp("key", result.key)
+  result.showcaseVisibility = jsonObj.toProfileShowcaseVisibility()
+
+proc toProfileShowcaseSaveSocialLink*(jsonObj: JsonNode): ProfileShowcaseSaveSocialLink =
+  result = ProfileShowcaseSaveSocialLink()
+  discard jsonObj.getProp("url", result.url)
+  discard jsonObj.getProp("text", result.text)
+  result.showcaseVisibility = jsonObj.toProfileShowcaseVisibility()
+
+proc toProfileShowcaseSaveData*(jsonObj: JsonNode): ProfileShowcaseSaveData =
+  result = ProfileShowcaseSaveData()
+  if jsonObj["communities"].kind != JNull:
+    for jsonMsg in jsonObj["communities"]:
+      result.communities.add(jsonMsg.toProfileShowcaseSaveEntry())
```

Comment:
These 3 lines get repeated over and over; some macro/inline proc? :)

Reply:
fixed